compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestBuildListener.java

```diff
@@ -173,13 +174,19 @@ private void setToManifest(TomlParser.ArrayValuesContext arrayValuesContext) {
      * @param key key specified in the header
      */
     private void addHeader(List<String> key) {
-        currentHeader = key.get(0);
-        if (key.size() > 1) {
-            StringJoiner joiner = new StringJoiner(".");
-            for (int i = 1; i < key.size(); i++) {
-                joiner.add(key.get(i));
+        String header = key.get(0);
```

Comment:
Why only the first value in the list is checked?

Reply:
The first value in the list will be the header. The ones that follow the header after the period will be names of dependencies/patches. 

Eg:  [dependencies."wso2/twitter"]
        version = "0.9.16"
Here the main header is "dependencies"